processor/samplingprocessor/tailsamplingprocessor/sampling/string_tag_probabilistic_filter.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2019, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sampling
+
+import (
+	"strconv"
+	"strings"
+
+	tracepb "github.com/census-instrumentation/opencensus-proto/gen-go/trace/v1"
+)
+
+type stringProbabilisticAttributeFilter struct {
+	key         string
+	values      map[string]bool
+	probability map[string]CountingSampler
+}
+
+var _ PolicyEvaluator = (*stringProbabilisticAttributeFilter)(nil)
+
+// NewStringProbabilisticAttributeFilter : NewStringAttributeFilter creates a policy evaluator that samples all traces with the given attribute in the given numeric range.
+func NewStringProbabilisticAttributeFilter(key string, values []string) PolicyEvaluator {
+	valuesMap := make(map[string]bool)
+	probabilityMap := make(map[string]CountingSampler)
+	for _, value := range values {
+		if value != "" {
+			v := strings.Split(value, "::")
+			valuesMap[v[0]] = true
+			f, _ := strconv.ParseFloat(v[1], 32)
+			probabilityMap[v[0]] = NewCountingSampler(float32(f))
+		}
+	}
+	return &stringProbabilisticAttributeFilter{
+		key:         key,
+		values:      valuesMap,
+		probability: probabilityMap,
```

Comment:
how about just using probabilityMap. If map holds the entry (sampler) then use the sampler. No need for valuesMap.

Reply:
Good Suggestion. Taken care. Thanks